src/main/scala/msgpack4z/MsgpackUnion.scala

```diff
@@ -243,7 +242,15 @@ object MsgpackUnion {
       case MsgType.BINARY =>
         new MsgpackBinary(unpacker.unpackBinary())
       case MsgType.EXTENDED =>
-        MsgpackExt // TODO
+        val ext = unpacker.unpackExtensionType()
+        val len = ext.getLength
+        val data = new Array[Byte](len)
+        var i = 0
+        while (i < len) {
+          data(i) = unpacker.unpackByte()
```

Comment:
I finally found the time to look at this again. You were right, the implementation was not correct. I replaced the loop by a call to `readPayload`.